### PR TITLE
Kr docs

### PR DESCRIPTION
**Pull request by zoomKoding**
## Status
**READY**

## Breaking Changes
NO

## Description
Korean docs

## Progress
- [ ] README
- [x] Cover page
- [ ] Sidebar
- Introduction
  - [x] Getting Started
  - [x] Why Bloc?
- Core Concepts
  - [x] bloc
  - [ ] flutter_bloc
- [x] Architecture
- [x] Testing
- [x] Naming Conventions
- [ ] FAQs
- Tutorials
  - Flutter
    - [ ] Counter
    - [ ] Timer
    - [ ] Infinite List
    - [ ] Login
    - [ ] Weather
    - [ ] Todos
    - [ ] Firebase Login
    - [ ] Firestore Todos
  - AngularDart
    - [ ] Counter
  - Flutter + AngularDart
    - [ ] Github Search
- Recipes
  - Flutter
    - [ ] Show SnackBar
    - [ ] Navigation
    - [ ] Bloc Access
- Tools
  - Extensions
    - [ ] IntelliJ
    - [ ] VSCode

### CODE REVIEW

**Comment by zoomKoding** [comment]
If you have any ideas, let me know!

**Comment by felangel** [comment]
@zoomKoding just to clarify you want to merge your changes as is and address the remaining items in separate PRs?

**Comment by zoomKoding** [comment]
@felangel Yes!

File: docs/ko-kr/_sidebar.md [hunk 1]
```diff
@@ -1,3 +1,46 @@
 - 소개
 
-  - [시작하기](ko-kr/gettingstarted.md)
+  - [Getting Started](/ko-kr/gettingstarted.md)
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: can we please also translate the text in the sidebar ("Getting Started", "Why Bloc?", etc...)?

File: docs/ko-kr/architecture.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# Architecture
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: Can we translate the headings as well please?

**Comment by zoomKoding** [comment]
@felangel I changed the headers and sidebars to Korean

File: docs/ko-kr/_sidebar.md [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+    - [flutter_bloc](/ko-kr/flutterbloccoreconcepts.md)
+  - [구조](/ko-kr/architecture.md)
+  - [테스트](/ko-kr/testing.md)
+  - [네이밍 컨벤션](/ko-kr/blocnamingconventions.md)
+  - [자주하는 질문](/ko-kr/faqs.md)
+
+- 튜토리얼
+
+  - Flutter
+    - [Counter](fluttercountertutorial.md)
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: can you please also translate these (Counter, TImer, Infinite List, etc...)?

File: docs/ko-kr/_sidebar.md [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+    - [Firestore Todos](flutterfirestoretodostutorial.md)
+  - AngularDart
+    - [Counter](angularcountertutorial.md)
+  - Flutter + AngularDart
+    - [Github Search](flutterangulargithubsearch.md)
+
+- Recipes
+
+  - Flutter
+    - [Show SnackBar](recipesfluttershowsnackbar.md)
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and these?

File: docs/ko-kr/_sidebar.md [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+    - [Weather](flutterweathertutorial.md)
+    - [Todos](fluttertodostutorial.md)
+    - [Firebase Login](flutterfirebaselogintutorial.md)
+    - [Firestore Todos](flutterfirestoretodostutorial.md)
+  - AngularDart
+    - [Counter](angularcountertutorial.md)
+  - Flutter + AngularDart
+    - [Github Search](flutterangulargithubsearch.md)
+
+- Recipes
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and Recipes

File: docs/ko-kr/_sidebar.md [hunk 6]
```diff
@@ -1,3 +1,46 @@
...
+
+  - Flutter
+    - [Show SnackBar](recipesfluttershowsnackbar.md)
+    - [Navigation](recipesflutternavigation.md)
+    - [Bloc Access](recipesflutterblocaccess.md)
+
+- Tools
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and Tools

File: docs/ko-kr/_sidebar.md [hunk 7]
```diff
@@ -1,3 +1,46 @@
...
+
+  - Extensions
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and Extensions

File: docs/ko-kr/_sidebar.md [hunk 8]
```diff
@@ -1,3 +1,46 @@
...
+
+    - [IntelliJ](blocintellijextension.md)
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and these

File: docs/ko-kr/_sidebar.md [hunk 9]
```diff
@@ -1,3 +1,46 @@
...
+    - [VSCode](blocvscodeextension.md)
+
+- API Reference
```
Comment (felangel) [inline docs/ko-kr/_sidebar.md]: and API Reference

File: docs/ko-kr/architecture.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
+- Data
+  - Data Provider
+  - Repository
+- Business Logic
+- Presentation
+
+먼저, (User Interface에서 가장 먼) 제일 낮은 layer에서 시작해서 presentation layer까지 올라갈 예정입니다.
+
+## Data Layer
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: can you please also translate Data Layer?

File: docs/ko-kr/architecture.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+> data layer의 역할은 소스로 부터 data를 찾고 조절하는 일입니다.
+
+data layer은 두 부분으로 나눌 수 있습니다:
+
+- Repository
+- Data Provider
+
+이 layer은 어플리케이션의 가장 낮은 layer으로 데이터베이스, 네트워크 요청, 비동기 데이터 소스와 상호작용합니다.
+
+### Data Provider
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: can you please also translate Data Provider?

File: docs/ko-kr/architecture.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+        final Data filteredData = _filterData(dataSetA, dataSetB);
+        return filteredData;
+    }
+}
+```
+
+위의 함수와 같이, repository layer는 여러 data provider와 상호작용할 수 있으며 결과값을 business logic layer에 전달하기 전에 data를 변모시키기도 합니다.
+
+## Bloc (Business Logic) Layer
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: can you please also translate Bloc (Business Logic) Layer?

File: docs/ko-kr/architecture.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+                yield Success(data);
+            } catch (error) {
+                yield Failure(error);
+            }
+        }
+    }
+}
+```
+
+### Bloc-to-Bloc Communication
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: can you please also translate Bloc-to-Bloc Communication?

File: docs/ko-kr/architecture.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+
+  @override
+  Future<void> close() {
+    otherBlocSubscription.cancel();
+    return super.close();
+  }
+}
+```
+
+## Presentation Layer
```
Comment (felangel) [inline docs/ko-kr/architecture.md]: can you please also translate Presentation Layer?

File: docs/ko-kr/blocnamingconventions.md [hunk 15]
```diff
@@ -0,0 +1,60 @@
+# 네이밍 컨벤션
+
+!> Naming convention을 따르는 것은 단순히 추천하는 사항이며 완전히 선택적입니다. 선호하시는 naming convention을 따르셔도 좋습니다. Example이나 문서에서 단순함과 간결함을 위해 naming convention을 따르지 않을 수도 있습니다. 이 convention은 많은 개발자가 있는 큰 프로젝트에 강력히 추천합니다.
+
+## Event Conventions
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+
+> Event는 bloc 관점에서는 이미 발생한 것들이기 때문에 **과거형**으로 작성해주세요.
+
+### Anatomy
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+
+`BlocSubject` + `Noun (optional)` + `Verb (event)`
+
+?> Inition load event는 다음과 같이 표기될 수 있습니다: `BlocSubject` + `Started`
+
+#### Examples
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+
+✅ **Good**
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+`CounterStarted`
+`CounterIncremented`
+`CounterDecremented`
+`CounterIncrementRetried`
+
+❌ **Bad**
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+
+`Initial`
+`CounterInitialized`
+`Increment`
+`DoIncrement`
+`IncrementCounter`
+
+## State Conventions
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+
+> State는 한 시점의 스냅샷이기 때문에 명사로 표현해주세요.
+
+### Anatomy
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 22]
```diff
@@ -0,0 +1,60 @@
...
+
+`BlocSubject` + `Verb (action)` + `State`
+
+?> `State`는 다음 중 하나입니다: `Initial` | `Success` | `Failure` | `InProgress`.
+초기 state는 다음과 같이 표기 될 수 있습니다: `BlocSubject` + `Initial`.
+
+#### Examples
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+
+✅ **Good**
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/blocnamingconventions.md [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+
+`CounterInitial`
+`CounterLoadInProgress`
+`CounterLoadSuccess`
+`CounterLoadFailure`
+
+❌ **Bad**
```
Comment (felangel) [inline docs/ko-kr/blocnamingconventions.md]: can you please also translate this?

File: docs/ko-kr/coreconcepts.md [hunk 25]
```diff
@@ -0,0 +1,323 @@
+# 주요 개념
+
+?> [bloc](https://pub.dev/packages/bloc)을 사용하기 전에 다음 내용을 잘 읽고 이해하기 바랍니다.
+
+Bloc을 어떻게 사용하는지 이해하기 위해서는 알아야 하는 몇가지 개념이 있습니다.
+
+다음 섹션에서 각 개념들을 자세히 이야기해보고 실제 어플리케이션인 계수기  만들고 있기 때문에, state는 간단합니다.앱에 적용시켜보도록 하겠습니다.
+
+## Events
```
Comment (felangel) [inline docs/ko-kr/coreconcepts.md]: can you please also translate this?

File: docs/ko-kr/coreconcepts.md [hunk 26]
```diff
@@ -0,0 +1,323 @@
...
+
+```dart
+enum CounterEvent { increment, decrement }
+```
+
+이 상황에서 event를 `enum`을 이용해서 나타냈지만, bloc에 정보를 전달해야 한다거나 더욱 복잡한 상황에서는 `class`를 사용해야할 수도 있습니다.
+
+이렇게 event를 처음으로 정의해보았습니다! 이 시점에서 우리는 아직 Bloc을 사용하지 않았고 아무런 마법도 일어나지 않습니다; 이 코드는 Plain Dart 코드입니다.
+
+## States
```
Comment (felangel) [inline docs/ko-kr/coreconcepts.md]: can you please also translate this?

File: docs/ko-kr/coreconcepts.md [hunk 27]
```diff
@@ -0,0 +1,323 @@
...
+
+여기까지, 우리 앱이 반응할 두개의 event를 정의했습니다: `CounterEvent.increment` 와 `CounterEvent.decrement`.
+
+이제 우리는 어떻게 우리 어플리케이션의 state를 나타낼지 정의해야 합니다.
+
+우리는 계수기를 만들고 있기 때문에, state는 매우 간단합니다: 계수기의 현재 숫자을 나타내는 정수면 됩니다.
+
+앞으로 state의 복잡한 예시를 보게 되겠지만, 우리 어플리케이션의 state를 표현하는데 이 정도의 매우 단순한 형태면 충분합니다.
+
+## Transitions
```
Comment (felangel) [inline docs/ko-kr/coreconcepts.md]: can you please also translate this?

File: docs/ko-kr/coreconcepts.md [hunk 28]
```diff
@@ -0,0 +1,323 @@
...
+{
+  "currentState": 0,
+  "event": "CounterEvent.increment",
+  "nextState": 1
+}
+```
+
+모든 state 변화는 기록되기 때문에, 매우 쉽게 계측하고, 모든 사용자 상호 작용 및 상태 변화를 한 곳에서 추적 할 수 있습니다. 뿐만 아니라, 이는 time-travel 디버깅과 같은 기능을 가능하게 합니다.
+
+## Streams
```
Comment (felangel) [inline docs/ko-kr/coreconcepts.md]: can you please also translate this?

File: docs/ko-kr/coreconcepts.md [hunk 29]
```diff
@@ -0,0 +1,323 @@
...
+```dart
+@override
+void onError(Object error, StackTrace stackTrace) {
+  print('$error, $stackTrace');
+}
+```
+
+이제 우리는 `onError`을 override했기 때문에, `Exception`이 발생함에 따라 무엇이든 할 수 있습니다.
+
+## BlocDelegat
```
Comment (felangel) [inline docs/ko-kr/coreconcepts.md]:
```suggestion
## BlocDelegate
```

File: docs/ko-kr/gettingstarted.md [hunk 30]
```diff
@@ -1,3 +1,66 @@
...
+
+이제 bloc을 설치해봅시다.
+
+!> `pubspec.yaml` 파일과 같은 directory에서 다음 명령어를 실행합니다.
+
+- Dart와 AngularDart의 경우에는, `pub get` 명령어를 실행합니다.
+
+- Flutter의 경우에는, `flutter packages get` 명령어를 실행합니다.
+
+## Import
```
Comment (felangel) [inline docs/ko-kr/gettingstarted.md]: can you please also translate this?

File: docs/ko-kr/testing.md [hunk 31]
```diff
@@ -0,0 +1,111 @@
+# Testing
```
Comment (felangel) [inline docs/ko-kr/testing.md]: can you please also translate this?

File: docs/ko-kr/whybloc.md [hunk 32]
```diff
@@ -0,0 +1,29 @@
+# Why Bloc?
```
Comment (felangel) [inline docs/ko-kr/whybloc.md]: can you please also translate this?

**Comment by felangel** [comment]
> @felangel I changed the headers and sidebars to Korean

Just had a couple of other places where I think we're missing translations. Let me know if it you don't think certain words need to be translated 👍

**Pull request merged**